app/src/main/java/me/hufman/androidautoidrive/music/controllers/SpotifyAppController.kt

```diff
@@ -623,23 +696,42 @@ class SpotifyAppController(context: Context, val remote: SpotifyAppRemote, val w
 			}.setErrorCallback {
 				deferred.complete(LinkedList())
 			}
+		} else if (isArtistDirectory(directory)) {
+			GlobalScope.launch(defaultDispatcher) {
+				val artistMediaId = directory.mediaId
+				val artistTopSongs = webApi.getArtistTopSongs(this@SpotifyAppController, artistMediaId) ?: emptyList()
+				if (artistTopSongs.isNotEmpty()) {
+					deferred.complete(artistTopSongs)
+				} else {
+					loadPaginatedItems(directory.toListItem(), { !deferred.isCancelled }) { results ->
+						deferred.complete(removeShufflePlayButtonMetadata(results, artistMediaId))
+					}
+				}
+			}
 		} else {
 			loadPaginatedItems(directory.toListItem(), { !deferred.isCancelled }) { results ->
-				deferred.complete(results.filterNot { it.title == SpotifyWebApi.LIKED_SONGS_PLAYLIST_NAME })
+				deferred.complete(results.filterNot { it.title == SpotifyWebApi.LIKED_SONGS_PLAYLIST_NAME || it.title == SpotifyWebApi.ARTIST_SONGS_PLAYLIST_NAME })
 			}
 		}
 		return deferred.await()
 	}
 
+	/**
+	 * Returns whether the supplied [MusicMetadata] directory is an "artist" type.
+	 */
+	private fun isArtistDirectory(directory: MusicMetadata?): Boolean {
```

Comment:
If this is only used in the browse function, I think it would be slightly cleaner if there was a small lambda defined directly inside the browse function:
```
fun browse() {
    val isArtistDirectory: (MusicMetadata?) -> Boolean = {it?.mediaId?.contains(":artists:") == true}
```
The flag could be inlined entirely and just plain evaluated at the top of function too.

Reply:
I agree, moved it into the browse function as an inline variable like your suggestion